# Session #5 Homework

In [1]:
import pickle
import requests

## Question 1
What's the version of pipenv you installed?  
2021.5.29

## Question 2
What's the first hash for scikit-learn you get in Pipfile.lock?  
sha256:121f78d6564000dc5e968394f45aac87981fcaaf2be40cfcd8f07b2baa1e1829

## Question 3
What's the probability that this customer is churning?

In [2]:
# Load the dict vectorizer
with open('dv.bin', 'rb') as dv_bin:
    dv = pickle.load(dv_bin)
dv_bin.close()

In [3]:
# Load the model
with open('model1.bin', 'rb') as model_bin:
    model = pickle.load(model_bin)
model_bin.close()

In [4]:
customer = {"contract": "two_year", "tenure": 12, "monthlycharges": 19.7}
X = dv.transform(customer)
y = model.predict_proba(X)[0,1]
y.round(3)

0.115

The probability of churning is 0.115 .

## Question 4
What's the probability that this customer is churning?

In [5]:
#  waitress-serve --listen 0.0.0.0:8080 predict:app

url = "http://localhost:8080/predict"
customer = {"contract": "two_year", "tenure": 1, "monthlycharges": 10}
requests.post(url, json=customer).json()

{'churn': True, 'churn_probability': 0.999}

The probabilityis 0.999 .

## Question 5

Docker image ID is f0f43f7bc6e0 .

## Question 6
What's the probability that this customer is churning?

In [12]:
url = "http://localhost:9696/predict"
customer = {"contract": "two_year", "tenure": 12, "monthlycharges": 10}
requests.post(url, json=customer).json()

{'churn': True, 'churn_probability': 0.728}

The probability is 0.728 .